payment-request/throw_if_not_active.https.html

```diff
@@ -0,0 +1,166 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>PaymentRequest show() throws if doc is not fully active</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<link rel="help" href="https://w3c.github.io/payment-request/#dom-paymentrequest-show()">
+<body>
+<script>
+const validMethod = Object.freeze({
+  supportedMethods: "basic-card",
+});
+const validMethods = Object.freeze([validMethod]);
+const validAmount = Object.freeze({
+  currency: "USD",
+  value: "5.00",
+});
+const validTotal = Object.freeze({
+  label: "Total due",
+  amount: validAmount,
+});
+const validDetails = Object.freeze({
+  total: validTotal,
+});
+
+function getLoadedPaymentRequest(iframe, url) {
+  return new Promise(resolve => {
+    iframe.addEventListener(
+      "load",
+      () => {
+        const { PaymentRequest } = iframe.contentWindow;
+        const request = new PaymentRequest(validMethods, validDetails);
+        resolve(request);
+      },
+      { once: true }
+    );
+    iframe.src = url;
+  });
+}
+
+promise_test(async t => {
+  // Check that PaymentRequests can be constructed.
+  new PaymentRequest(validMethods, validDetails);
+  const iframe = document.createElement("iframe");
+  iframe.allowPaymentRequest = true;
+  document.body.appendChild(iframe);
+
+  // We first got to page1.html, grab a PaymentRequest instance.
+  const request1 = await getLoadedPaymentRequest(
+    iframe,
+    "/payment-request/resources/page1.html"
+  );
+
+  // We navigate the iframe again, putting request1's document into an inactive state.
+  const request2 = await getLoadedPaymentRequest(
+    iframe,
+    "/payment-request/resources/page2.html"
+  );
+
+  // Now, request1's relevant global object's document is no longer active.
+  // So, call .show(), and make sure it rejects appropriately.
+  await promise_rejects(
+    t,
+    "AbortError",
+    request1.show(),
+    "Inactive document, so must throw AbortError"
+  );
+
+  // request2 has an active document tho, so confirm it's working as expected:
+  request2.show();
+  await request2.abort();
+  await promise_rejects(
+    t,
+    "InvalidStateError",
+    request2.show(),
+    "Abort already called, so InvalidStateError"
+  );
+  // We are done, so clean up.
+  iframe.remove();
+}, "PaymentRequest.show() aborts if the document is not active");
+
+promise_test(async t => {
+  // check that PaymentRequests can be constructed.
+  new PaymentRequest(validMethods, validDetails);
+  // We create two iframes, and nest them. Wait for both of them to load.
+  const iframe = document.createElement("iframe");
+  iframe.allowPaymentRequest = true;
+  document.body.appendChild(iframe);
+  // We get the request from the inner-most iframe.
+  const request = await new Promise(resolve => {
```

Comment:
Can't you use your super-nice helper here?

```js
const request1 = await getLoadedPaymentRequest(iframe, "/payment-request/resources/page1.html");

const innerIframe = iframe.contentDocument.createElement("iframe");
innerIframe.allowPaymentRequest = true;
iframe.contentDocument.body.appendChild(innerIframe);

const request2 = await getLoadedPaymentRequest(innerIframe, "/payment-request/resources/page2.html");
```

? If I'm missing something, I'm still hopeful that you can create some variation on your helper that does work, since it's so nice.

Reply:
Will add tomorrow :( Ran out of time today.